cmd/launcher/watchdog/controller_windows.go

```diff
@@ -0,0 +1,340 @@
+
+		return
+	}
+
+	// we're enabling the watchdog - first check if we've already installed the service
+	// there are three potential paths here:
+	// 1. service did not previously exist, proceed with clean installation
+	existingService, err := serviceManager.OpenService(launcherWatchdogServiceName)
+	if err != nil && err.Error() == serviceDoesNotExistError {
+		if err = wc.installService(serviceManager); err != nil {
+			wc.slogger.Log(ctx, slog.LevelError,
+				"encountered error installing watchdog service",
+				"err", err,
+			)
+		}
+
+		return
+	}
+
+	// 2. we are unable to check the current status of the service,
+	// this is the least likely option and there's nothing we can do here so log and return
+	if err != nil {
+		wc.slogger.Log(ctx, slog.LevelWarn,
+			"encountered error checking for watchdog service, unable to proceed with enabling",
+			"err", err,
+		)
+
+		return
+	}
+
+	// 3. The watchdog service already exists on this device. Here we just restart it to ensure it is
+	// running on the latest launcher code
+	defer existingService.Close()
+	if err = wc.restartService(existingService); err != nil {
+		wc.slogger.Log(ctx, slog.LevelError,
+			"failure attempting to restart watchdog service",
+			"err", err,
+		)
+	}
+}
+
+func (wc *WatchdogController) installService(serviceManager *mgr.Mgr) error {
+	ctx := context.TODO()
+	currentExe, err := os.Executable()
+	if err != nil {
+		wc.slogger.Log(ctx, slog.LevelError,
+			"installing launcher watchdog, unable to collect current executable path",
+			"err", err,
+		)
+
+		return fmt.Errorf("collecting current executable path: %w", err)
+	}
+
+	svcMgrConf := mgr.Config{
+		DisplayName:  launcherWatchdogServiceName,
+		Description:  "The Kolide Launcher Watchdog Service",
+		StartType:    mgr.StartAutomatic,
+		ErrorControl: mgr.ErrorNormal,
+		// no reason to rush start for this service, we should wait until after
+		// the main launcher service has attempted to start anyway
+		DelayedAutoStart: true,
+	}
+
+	serviceArgs := []string{"watchdog"}
+	// add any original service arguments from the main launcher service invocation (currently running)
+	// this is likely just a pointer to the launcher.flags file but we want to ensure that the watchdog service
+	// has insight into the same options for early service configuration, logging, etc.
+	serviceArgs = append(serviceArgs, os.Args[2:]...)
+
+	restartService, err := serviceManager.CreateService(
+		launcherWatchdogServiceName,
+		currentExe,
+		svcMgrConf,
+		serviceArgs...,
+	)
+
+	if err != nil { // no point moving forward if we can't create the service
+		return err
+	}
+
+	defer restartService.Close()
+
+	// set recovery actions - always restart after a 5 second delay
+	recoveryActions := []mgr.RecoveryAction{
+		{
+			Type:  mgr.ServiceRestart,
+			Delay: 5 * time.Second,
+		},
+	}
+
+	if err = restartService.SetRecoveryActions(recoveryActions, 10800); err != nil {
```

Comment:
because this looks like a fun, unimportant thing to quibble over, I would go with

`const serviceResetPeriodSeconds = 3 * 60 * 60 // 3 hours in seconds`